app.js

```diff
@@ -1,24 +1,33 @@
 const express = require("express");
 const app = express();
-const getTopics = require("../db/controller/topics-controller");
-const endpoints = require("../endpoints.json");
+const getTopics = require("./controller/topics-controller");
+const getArticles = require("./controller/articles-controller");
+const endpoints = require("./endpoints.json");
 
 app.use(express.json());
 
 app.get("/api/topics", getTopics);
+app.get(`/api/articles/:article_id`, getArticles);
 app.get(`/api`, (req, res, next) => {
   res.status(200).send(endpoints);
-})
+});
+
+app.all("*", (req, res, next) => {
+  res.status(404).send({ message: "path not found!" });
+  next(err);
+});
 
 app.use((err, req, res, next) => {
-  if (err.status && err.message) {
-    res.status(err.status).send({ message: err.message });
+  if (err.code === `22P02`) {
+    res.status(400).send({ message: "invalid id type!" });
   }
   next(err);
 });
 
-app.all("*", (req, res, next) => {
-  res.status(404).send({ message: "path not found!" });
+app.use((err, req, res, next) => {
+  if (err.status && err.message) {
+    res.status(err.status).send({ message: err.message });
+  }
```

Comment:
I recommend you to add another error handling middleware at the end to handle 500's and send an appropriate response to the user in case of an internal server error.